### PR TITLE
Add support for CocoaPods

### PR DESCRIPTION
**Pull request by mlch911**

### CODE REVIEW

File: SaferContinuation.podspec
Comment (mredig) [inline SaferContinuation.podspec:1]: I've been fortunate enough to not have to use CocoaPods very much, but as a consequence don't remember much about how it works.

How hard will it be to maintain this going forward, as someone who doesn't use it?

File: Sources/SaferContinuation/SaferContinuation.swift [hunk 1]
```diff
@@ -51,9 +51,22 @@ final public class SaferContinuation<C: Continuation & Sendable>: @unchecked Sen
...
 	 */
-	public init(_ continuation: C, isFatal: SaferContinuation<UnsafeContinuation<Void, Error>>.FatalityOptions = false, timeout: TimeInterval? = nil, delayCheckInterval: TimeInterval? = 3, file: StaticString = #file, line: Int = #line, function: StaticString = #function, context: Any? = nil) {
+	public init(_ continuation: C,
+                isFatal: SaferContinuation<UnsafeContinuation<Void, Error>>.FatalityOptions? = nil,
+                timeout: TimeInterval? = nil,
+                delayCheckInterval: TimeInterval? = 3,
+                file: StaticString = #file,
+                line: Int = #line, 
+                function: StaticString = #function,
+                context: Any? = nil) {
```
Comment (mredig) [inline Sources/SaferContinuation/SaferContinuation.swift]: (this comment applies to the rest of the PR as well) this is a tabs indentation project. Please fix the indentation to reflect as such.

Also, please place a newline before `_` and after `nil` so the resulting indentation looks like

```swift
	func init(
		_ continuation: C,

		...

		context: Any? = nil
	) {
	// implementation

	}
```

I need to write up a style guide, so I apologize for this style not being apparent earlier.

File: Sources/SaferContinuation/SaferContinuation.swift [hunk 2]
```diff
@@ -51,9 +51,22 @@ final public class SaferContinuation<C: Continuation & Sendable>: @unchecked Sen
...
+#if DEBUG
+        let defaultFatalOptions: SaferContinuation<UnsafeContinuation<Void, Error>>.FatalityOptions = true
+#else
+        let defaultFatalOptions: SaferContinuation<UnsafeContinuation<Void, Error>>.FatalityOptions = false
+#endif
+
```
Comment (mredig) [inline Sources/SaferContinuation/SaferContinuation.swift]: What is the justification for changing this behavior?

I'm not saying you're wrong, I just see benefits to both approaches, I want to see if there are any real world scenarios I hadn't thought of that would benefit from this approach, especially before changing existing behavior.

File: Sources/SaferContinuation/SaferContinuation.swift [hunk 3]
```diff
@@ -19,7 +19,7 @@ final public class SaferContinuation<C: Continuation & Sendable>: @unchecked Sen
 
 	let isFatal: SaferContinuation<UnsafeContinuation<Void, Error>>.FatalityOptions
 
-	private var hasRun = false
+	public private(set) var hasRun = false
```
Comment (mredig) [inline Sources/SaferContinuation/SaferContinuation.swift:22]: Good idea!

File: Sources/SaferContinuation/SaferContinuation.swift [hunk 4]
```diff
@@ -235,4 +248,14 @@ extension SaferContinuation where C == UnsafeContinuation<Void, Error> {
...
+    
+    public func resume() {
+        resume(returning: ())
+    }
+}
+
+extension SaferContinuation where C == CheckedContinuation<Void, Error> {
+    public func resume() {
+        resume(returning: ())
+    }
```
Comment (mredig) [inline Sources/SaferContinuation/SaferContinuation.swift]: 🔥

File: Sources/SaferContinuation/SaferContinuationError.swift [hunk 5]
```diff
@@ -2,14 +2,14 @@ import Foundation
 
 public enum SafeContinuationError: Error {
 	/// internally used by safercontinuation. cannot be thrown by the continuation as that would cause bugs to have the continuation resume multiple times (which is exactly what we are avoiding here)
-	case alreadyRun(file: StaticString, line: Int, function: StaticString, context: Any?)
+    case alreadyRun(file: StaticString = #file, line: Int = #line, function: StaticString = #function, context: Any? = nil)
```
Comment (mredig) [inline Sources/SaferContinuation/SaferContinuationError.swift]: Indentation on this line

**Review by mredig — CHANGES_REQUESTED** [review]
Thanks for the contribution! I have a few small tweaks and a question in inline comments.

File: Sources/SaferContinuation/SaferContinuation.swift (hunk 2 above)
Comment (mlch911) [reply Sources/SaferContinuation/SaferContinuation.swift]: With this, we can find some potential problems during development and debugging, and there will be no fatalError at release.

File: Sources/SaferContinuation/SaferContinuation.swift (hunk 1 above)
Comment (mlch911) [reply Sources/SaferContinuation/SaferContinuation.swift]: done.

File: SaferContinuation.podspec
Comment (mlch911) [reply SaferContinuation.podspec:1]: There is no big difference between CocoaPods and SPM, especially for the current simple repo. If you don't have any major changes of the repo structure, you just need to update the version number in the podspec every time.

Comment (mredig) [reply SaferContinuation.podspec:1]: Sounds good - I think I can manage that :)

File: Sources/SaferContinuation/SaferContinuation.swift (hunk 2 above)
Comment (mredig) [reply Sources/SaferContinuation/SaferContinuation.swift]: My apologies. I had the model inverted in my mind and was assuming that I previously had it default to crash on error, so I thought it was being modified to step down security, not up. This is fine :)

File: Sources/SaferContinuation/SaferContinuationError.swift [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
 
-	/// thrown by safer continuation when the continuation is deinited and has never been called
-	case continuationNeverCompleted(file: StaticString, line: Int, function: StaticString, context: Any?)
+    /// thrown by safer continuation when the continuation is deinited and has never been called
+    case continuationNeverCompleted(file: StaticString = #file, line: Int = #line, function: StaticString = #function, context: Any? = nil)
 
-	/// thrown by safer continuation when the continuation doesn't fire within the timeout period
-	case timeoutMet(file: StaticString, line: Int, function: StaticString, context: Any?)
+    /// thrown by safer continuation when the continuation doesn't fire within the timeout period
+    case timeoutMet(file: StaticString = #file, line: Int = #line, function: StaticString = #function, context: Any? = nil)
```
Comment (mredig) [inline Sources/SaferContinuation/SaferContinuationError.swift]: I meant they should be tabs, not spaces.

However, I now remember why I hadn't done this in the first place: these values are not about the file/line/function/context of where the error is thrown, but about where the continuation is created, hence why the values are forwarded from the continuation properties. It was correct to not have default values in the first place.

**Review by mredig — CHANGES_REQUESTED** [review]
One more thing

File: Sources/SaferContinuation/SaferContinuationError.swift (hunk 6 above)
Comment (mlch911) [reply Sources/SaferContinuation/SaferContinuationError.swift]: The code indentation is something I missed.

These default values are useful. When we use these errors outside, at our real code, these default values are really convenient. When use these errors inside this project, we should pass the value from outside.

Comment (mredig) [reply Sources/SaferContinuation/SaferContinuationError.swift]: These errors should only be initialized internally. External methods will receive them when thrown, but shouldn't be creating them. And again, the metadata of the file/line/function is to help identify which *continuation* is having issues, not the error itself... So the default arguments to get the error's location is irrelevant, regardless.

Comment (mlch911) [reply Sources/SaferContinuation/SaferContinuationError.swift]: Done.

**Review by mredig — APPROVED** [review]

**Comment by mredig** [comment]
Thank you! Looks great :)

**Pull request merged**